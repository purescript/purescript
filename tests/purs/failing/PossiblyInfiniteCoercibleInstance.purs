-- @shouldFailWith PossiblyInfiniteCoercibleInstance
module Main where

import Safe.Coerce (coerce)

newtype N a = N (a -> N a)

infinite :: forall a b. N a -> N b
infinite = coerce
