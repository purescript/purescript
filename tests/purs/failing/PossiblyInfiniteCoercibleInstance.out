Error found:
in module [33mMain[0m
at tests/purs/failing/PossiblyInfiniteCoercibleInstance.purs:9:12 - 9:18 (line 9, column 12 - line 9, column 18)

  A [33mCoercible[0m instance is possibly infinite.

while solving type class constraint
[33m                              [0m
[33m  Prim.Coerce.Coercible (N a0)[0m
[33m                        (N b1)[0m
[33m                              [0m
while checking that type [33mforall (a :: Type) (b :: Type). Coercible @Type a b => a -> b[0m
  is at least as general as type [33mN a0 -> N b1[0m
while checking that expression [33mcoerce[0m
  has type [33mN a0 -> N b1[0m
in value declaration [33minfinite[0m

where [33ma0[0m is a rigid type variable
        bound at (line 9, column 12 - line 9, column 18)
      [33mb1[0m is a rigid type variable
        bound at (line 9, column 12 - line 9, column 18)

See https://github.com/purescript/documentation/blob/master/errors/PossiblyInfiniteCoercibleInstance.md for more information,
or to contribute content related to this error.

